src/conn/conn_ckpt.c

```diff
@@ -78,14 +94,20 @@ __ckpt_server(void *arg)
 	conn = S2C(session);
 	wt_session = (WT_SESSION *)session;
 
-	while (F_ISSET(conn, WT_CONN_SERVER_RUN) &&
-	    F_ISSET(conn, WT_CONN_SERVER_CHECKPOINT)) {
+	for (;;) {
+		if (!__ckpt_server_run_chk(session))
+			break;
```

Comment:
I see that everywhere now that I pull the branch and look at the code instead of the diff.  All of the 4-5 functions are identically structured. What about encapsulating in a function:
`bool __wt_server_wait_and_run(same 4 args as cond_wait)` and that then makes it caller look like this (statlog for example):
```
while (__wt_server_wait_and_run(session, conn->stat_cond, conn->stat_usecs, __statlog_server_run_chk)
    if (WT_STAT_ENABLED(session))
        WT_ERR(__statlog_log_one(session, ...);
```
```